.github/workflows/update-psl.yml

```diff
@@ -0,0 +1,48 @@
+name: Update PSL
+
+on:
+  workflow_dispatch:
+  schedule:
+    # Run monthly at a random (https://xkcd.com/221/) time.
+    - cron: '46 8 1 * *'
+
+jobs:
+  create-pr:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v3
+        with:
+          go-version-file: 'go.mod'
```

Comment:
> This is a bit of an odd way to specify the Go version to use, and [doesn't seem to accord 100% with the semantics of go.mod](https://go.dev/ref/mod#go-mod-file-go).

That page says "A `go` directive indicates that a module was written assuming the semantics of a given version of Go." In my opinion, every time we remove an older Go version from our testing matrix, we are no longer writing our module assuming the semantics of that version. This is especially true in the most recent transition -- we removed go1.17 from our testing matrix, and now we can write code with generics, so having go1.18 in our go.mod is both necessary and correct. I would interpret the `go` directive as "the minimum toolchain version necessary to interact with this repo", which I think makes it the perfect candidate for setting up the Go toolchain to interact with the repo in this workflow.

> Does this solve a specific problem?

It solves the problem of "having to update the Go version in multiple places when we want/need to". I can't think of any reason why we'd purposefully want this workflow to use a different version of Go than what's in the go.mod file, so it just saves us one update.

> If we do need to specify the Go version to use via this method, I'd like to apply it consistently to all the places where we specify Go version.

Totally agreed. But we don't use the setup-go action in any of our other workflows, because the other workflows execute their actions inside the boulder-tools container, so there's nowhere else in this repo to update.

Reply:
What about running the `go` commands via boulder-tools? `docker-compose run boulder go get ...`; `docker-compose run boulder go mod vender`, etc. That would ensure we run the exact version in our boulder-tools. That's important because we have the `gomod-vendor` test phase that ensures things are vendored correctly, and differences would cause that phase to fail.

Reply:
If we run via `docker-compose run`, we get the default BOULDER_TOOLS_TAG from docker-compose.yml, which we keep up to date.

Reply:
The latest patchset now uses docker-compose to do this.

Weirdly I couldn't get `docker-compose run netaccess sh -c "go get ... && go mod tidy && go mod vendor"` to work; for some reason the Go command bails out and prints its usage when run inside `sh -c`. But regardless, running three separate commands works at the cost of marginally higher startup time for each command.